pyproject.toml

```diff
@@ -36,6 +36,7 @@ dependencies = [
 
 [project.optional-dependencies]
 test = ["pytest","pytest-mock"]
+docs = ["Sphinx==7.1.2","sphinx-rtd-theme==2.0.0","myst-parser==3.0.1"]
```

Comment:
@carlstreeter-quic @quic-jhugo , the packages will go as additional installation for docs only.
for someone to build docs, they can use `pip install -e.[docs]`. Please review.

Reply:
@carlstreeter-quic @quic-jhugo ,we will be merging this PR early next week. `(ETA August 13th 2024)`
Please review and let us know if you have any queries. Thanks